### ISSUE TITLE
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process:

### ISSUE THREAD

**Issue by markgahagan1** [stefan-jansen/zipline-reloaded#169]
Dear Zipline Maintainers,

Before I tell you about my issue, let me describe my environment:

* Operating System: Windows 11
* Python Version: Anaconda python environment 3.8.16
* Python Bitness: 64
* How did you install Zipline: As per pypy instructions for zipline-norgatedata
* Python packages: # Name                    Version                   Build  Channel
_anaconda_depends         2022.10                  py39_2
_ipyw_jlab_nb_ext_conf    0.1.0            py39haa95532_0
alabaster                 0.7.12             pyhd3eb1b0_0
anaconda                  custom                   py39_1
anaconda-client           1.11.0           py39haa95532_0
anaconda-navigator        2.3.2            py39haa95532_0
anyio                     3.5.0            py39haa95532_0
appdirs                   1.4.4              pyhd3eb1b0_0
argon2-cffi               21.3.0             pyhd3eb1b0_0
argon2-cffi-bindings      21.2.0           py39h2bbff1b_0
arrow                     1.2.3            py39haa95532_0
astroid                   2.11.7           py39haa95532_0
astropy                   5.1              py39h080aedc_0
atomicwrites              1.4.0                      py_0
attrs                     22.1.0           py39haa95532_0
automat                   20.2.0                     py_0
autopep8                  1.6.0              pyhd3eb1b0_1
babel                     2.11.0           py39haa95532_0
backcall                  0.2.0              pyhd3eb1b0_0
backports                 1.1                pyhd3eb1b0_0
backports.functools_lru_cache 1.6.4              pyhd3eb1b0_0
backports.tempfile        1.0                pyhd3eb1b0_1
backports.weakref         1.0.post1                  py_1
bcrypt                    3.2.0            py39h2bbff1b_1
beautifulsoup4            4.11.1           py39haa95532_0
binaryornot               0.4.4              pyhd3eb1b0_1
bitarray                  2.5.1            py39h2bbff1b_0
bkcharts                  0.2              py39haa95532_1
black                     22.6.0           py39haa95532_0
blas                      1.0                         mkl
bleach                    4.1.0              pyhd3eb1b0_0
blosc                     1.21.3               h6c2663c_0
bokeh                     2.4.3            py39haa95532_0
boto3                     1.24.28          py39haa95532_0
botocore                  1.27.59          py39haa95532_0
bottleneck                1.3.5            py39h080aedc_0
brotli                    1.0.9                h2bbff1b_7
brotli-bin                1.0.9                h2bbff1b_7
brotlipy                  0.7.0           py39h2bbff1b_1003
bzip2                     1.0.8                he774522_0
ca-certificates           2023.01.10           haa95532_0
certifi                   2022.12.7        py39haa95532_0
cffi                      1.15.1           py39h2bbff1b_3
cfitsio                   3.470                h2bbff1b_7
chardet                   4.0.0           py39haa95532_1003
charls                    2.2.0                h6c2663c_0
charset-normalizer        2.0.4              pyhd3eb1b0_0
click                     8.0.4            py39haa95532_0
cloudpickle               2.0.0              pyhd3eb1b0_0
clyent                    1.2.2            py39haa95532_1
colorama                  0.4.6            py39haa95532_0
colorcet                  3.0.1            py39haa95532_0
comm                      0.1.2            py39haa95532_0
comtypes                  1.1.10          py39haa95532_1002
conda                     23.1.0           py39haa95532_0
conda-build               3.22.0           py39hcbf5309_2    conda-forge
conda-content-trust       0.1.3            py39haa95532_0
conda-env                 2.6.0                haa95532_1
conda-package-handling    2.0.2            py39haa95532_0
conda-package-streaming   0.7.0            py39haa95532_0
conda-repo-cli            1.0.27           py39haa95532_0
conda-token               0.4.0              pyhd3eb1b0_0
conda-verify              3.4.2                      py_1
console_shortcut          0.1.1                         4
constantly                15.1.0             pyh2b92418_0
contourpy                 1.0.5            py39h59b6b97_0
cookiecutter              1.7.3              pyhd3eb1b0_0
cryptography              38.0.4           py39h21b164f_0
cssselect                 1.1.0              pyhd3eb1b0_0
curl                      7.87.0               h2bbff1b_0
cycler                    0.11.0             pyhd3eb1b0_0
cython                    0.29.33          py39hd77b12b_0
cytoolz                   0.12.0           py39h2bbff1b_0
daal4py                   2021.6.0         py39h757b272_1
dal                       2021.6.0           h59b6b97_874
dask                      2022.5.0         py39haa95532_0
dask-core                 2022.5.0         py39haa95532_0
datashader                0.14.4           py39haa95532_0
datashape                 0.5.4            py39haa95532_1
debugpy                   1.5.1            py39hd77b12b_0
decorator                 5.1.1              pyhd3eb1b0_0
defusedxml                0.7.1              pyhd3eb1b0_0
diff-match-patch          20200713           pyhd3eb1b0_0
dill                      0.3.6            py39haa95532_0
distributed               2022.5.0         py39haa95532_0
docutils                  0.18.1           py39haa95532_3
entrypoints               0.4              py39haa95532_0
et_xmlfile                1.1.0            py39haa95532_0
fftw                      3.3.9                h2bbff1b_1
filelock                  3.9.0            py39haa95532_0
flake8                    4.0.1              pyhd3eb1b0_1
flask                     2.2.2            py39haa95532_0
flit-core                 3.6.0              pyhd3eb1b0_0
fmt                       9.1.0                h6d14046_0
fonttools                 4.25.0             pyhd3eb1b0_0
freetype                  2.12.1               ha860e81_0
fsspec                    2022.11.0        py39haa95532_0
future                    0.18.2           py39haa95532_1
gensim                    4.3.0            py39hd77b12b_0
gettext                   0.21.1               h5728263_0    conda-forge
giflib                    5.2.1                h8cc25b3_1
glib                      2.74.1               h12be248_1    conda-forge
glib-tools                2.74.1               h12be248_1    conda-forge
glob2                     0.7                pyhd3eb1b0_0
greenlet                  2.0.1            py39hd77b12b_0
gst-plugins-base          1.21.3               h001b923_1    conda-forge
gstreamer                 1.21.3               h6b5321d_1    conda-forge
h5py                      3.7.0            py39h3de5c98_0
hdf5                      1.10.6               h1756f20_1
heapdict                  1.0.1              pyhd3eb1b0_0
holoviews                 1.15.4           py39haa95532_0
hvplot                    0.8.2            py39haa95532_0
hyperlink                 21.0.0             pyhd3eb1b0_0
icc_rt                    2022.1.0             h6049295_2
icu                       70.1                 h0e60522_0    conda-forge
idna                      3.4              py39haa95532_0
imagecodecs               2021.8.26        py39hc0a7faf_1
imageio                   2.19.3           py39haa95532_0
imagesize                 1.4.1            py39haa95532_0
importlib-metadata        4.11.3           py39haa95532_0
importlib_metadata        4.11.3               hd3eb1b0_0
incremental               21.3.0             pyhd3eb1b0_0
inflection                0.5.1            py39haa95532_0
iniconfig                 1.1.1              pyhd3eb1b0_0
intake                    0.6.7            py39haa95532_0
intel-openmp              2021.4.0          haa95532_3556
intervaltree              3.1.0              pyhd3eb1b0_0
ipykernel                 6.19.2           py39hd4e2768_0
ipython                   7.31.1           py39haa95532_1
ipython_genutils          0.2.0              pyhd3eb1b0_1
ipywidgets                7.6.5              pyhd3eb1b0_1
isort                     5.9.3              pyhd3eb1b0_0
itemadapter               0.3.0              pyhd3eb1b0_0
itemloaders               1.0.4              pyhd3eb1b0_1
itsdangerous              2.0.1              pyhd3eb1b0_0
jdcal                     1.4.1              pyhd3eb1b0_0
jedi                      0.18.1           py39haa95532_1
jellyfish                 0.9.0            py39h2bbff1b_0
jinja2                    3.1.2            py39haa95532_0
jinja2-time               0.2.0              pyhd3eb1b0_3
jmespath                  0.10.0             pyhd3eb1b0_0
joblib                    1.1.1            py39haa95532_0
jpeg                      9e                   h2bbff1b_0
jq                        1.6                  haa95532_1
json5                     0.9.6              pyhd3eb1b0_0
jsonschema                4.17.3           py39haa95532_0
jupyter                   1.0.0            py39haa95532_8
jupyter_client            7.4.9            py39haa95532_0
jupyter_console           6.4.4            py39haa95532_0
jupyter_core              5.2.0            py39haa95532_0
jupyter_server            1.23.4           py39haa95532_0
jupyterlab                3.5.3            py39haa95532_0
jupyterlab_pygments       0.1.2                      py_0
jupyterlab_server         2.16.5           py39haa95532_0
jupyterlab_widgets        1.0.0              pyhd3eb1b0_1
keyring                   23.4.0           py39haa95532_0
kiwisolver                1.4.4            py39hd77b12b_0
krb5                      1.20.1               heb0366b_0    conda-forge
lazy-object-proxy         1.6.0            py39h2bbff1b_0
lcms2                     2.12                 h83e58a3_0
lerc                      3.0                  hd77b12b_0
libaec                    1.0.4                h33f27b4_1
libarchive                3.6.2                h27c7867_0    conda-forge
libbrotlicommon           1.0.9                h2bbff1b_7
libbrotlidec              1.0.9                h2bbff1b_7
libbrotlienc              1.0.9                h2bbff1b_7
libclang                  15.0.7          default_h77d9078_1    conda-forge
libclang13                15.0.7          default_h77d9078_1    conda-forge
libcurl                   7.87.0               h86230a5_0
libdeflate                1.8                  h2bbff1b_5
libffi                    3.4.2                hd77b12b_6
libglib                   2.74.1               he8f3873_1    conda-forge
libiconv                  1.17                 h8ffe710_0    conda-forge
liblief                   0.12.3               hd77b12b_0
libmamba                  1.3.1                h8a7d157_1    conda-forge
libmambapy                1.3.1            py39h608c70e_1    conda-forge
libogg                    1.3.5                h2bbff1b_1
libpng                    1.6.39               h19919ed_0    conda-forge
libsodium                 1.0.18               h62dcd97_0
libsolv                   0.7.23               h12be248_0    conda-forge
libspatialindex           1.9.3                h6c2663c_0
libsqlite                 3.40.0               hcfcfb64_0    conda-forge
libssh2                   1.10.0               h9a1e1f7_3    conda-forge
libtiff                   4.5.0                h6c2663c_1
libvorbis                 1.3.7                he774522_0
libwebp                   1.2.4                h2bbff1b_0
libwebp-base              1.2.4                h2bbff1b_0
libxml2                   2.10.3               hc3477c8_0    conda-forge
libxslt                   1.1.37               h0192164_0    conda-forge
libzlib                   1.2.13               hcfcfb64_4    conda-forge
libzopfli                 1.0.3                ha925a31_0
llvmlite                  0.39.1           py39h23ce68f_0
locket                    1.0.0            py39haa95532_0
lxml                      4.9.2            py39h0942119_0    conda-forge
lz4                       3.1.3            py39h2bbff1b_0
lz4-c                     1.9.4                h2bbff1b_0
lzo                       2.10                 he774522_2
m2-msys2-runtime          2.5.0.17080.65c939c               3    conda-forge
m2-patch                  2.7.5                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2
mamba                     1.3.1            py39hca8391b_1    conda-forge
markdown                  3.4.1            py39haa95532_0
markupsafe                2.1.1            py39h2bbff1b_0
matplotlib                3.6.2            py39haa95532_0
matplotlib-base           3.6.2            py39h1094b8e_0
matplotlib-inline         0.1.6            py39haa95532_0
mccabe                    0.7.0              pyhd3eb1b0_0
menuinst                  1.4.18           py39h59b6b97_0
mistune                   0.8.4           py39h2bbff1b_1000
mkl                       2021.4.0           haa95532_640
mkl-service               2.4.0            py39h2bbff1b_0
mkl_fft                   1.3.1            py39h277e83a_0
mkl_random                1.2.2            py39hf11a4ad_0
mock                      4.0.3              pyhd3eb1b0_0
mpmath                    1.2.1            py39haa95532_0
msgpack-python            1.0.3            py39h59b6b97_0  
msys2-conda-epoch         20160418                      1
multipledispatch          0.6.0            py39haa95532_0
munkres                   1.1.4                      py_0
mypy_extensions           0.4.3            py39haa95532_1
navigator-updater         0.2.1                    py39_1
nbclassic                 0.4.8            py39haa95532_0
nbclient                  0.5.13           py39haa95532_0
nbconvert                 6.5.4            py39haa95532_0
nbformat                  5.7.0            py39haa95532_0
nest-asyncio              1.5.6            py39haa95532_0
networkx                  2.8.4            py39haa95532_0
nltk                      3.7                pyhd3eb1b0_0
nose                      1.3.7           pyhd3eb1b0_1008
notebook                  6.5.2            py39haa95532_0
notebook-shim             0.2.2            py39haa95532_0
numba                     0.56.4           py39hf11a4ad_0
numexpr                   2.8.4            py39h5b0cc5e_0
numpy                     1.23.5           py39h3b20f71_0
numpy-base                1.23.5           py39h4da318b_0
numpydoc                  1.5.0            py39haa95532_0
olefile                   0.46               pyhd3eb1b0_0
openjpeg                  2.4.0                h4fc8c34_0
openpyxl                  3.0.10           py39h2bbff1b_0
openssl                   3.0.8                hcfcfb64_0    conda-forge
packaging                 22.0             py39haa95532_0
pandas                    1.5.2            py39hf11a4ad_0
pandocfilters             1.5.0              pyhd3eb1b0_0
panel                     0.14.3           py39haa95532_0
param                     1.12.3           py39haa95532_0
paramiko                  2.8.1              pyhd3eb1b0_0
parsel                    1.6.0            py39haa95532_0
parso                     0.8.3              pyhd3eb1b0_0
partd                     1.2.0              pyhd3eb1b0_1
pathlib                   1.0.1              pyhd3eb1b0_1
pathspec                  0.10.3           py39haa95532_0
patsy                     0.5.3            py39haa95532_0
pcre2                     10.40                h17e33f8_0    conda-forge
pep8                      1.7.1            py39haa95532_1
pexpect                   4.8.0              pyhd3eb1b0_3
pickleshare               0.7.5           pyhd3eb1b0_1003
pillow                    9.3.0            py39hd77b12b_2
pip                       22.3.1           py39haa95532_0
pkginfo                   1.8.3            py39haa95532_0
platformdirs              2.5.2            py39haa95532_0
plotly                    5.9.0            py39haa95532_0
pluggy                    1.0.0            py39haa95532_1
ply                       3.11             py39haa95532_0
pooch                     1.4.0              pyhd3eb1b0_0
powershell_shortcut       0.0.1                         3
poyo                      0.5.0              pyhd3eb1b0_0
prometheus_client         0.14.1           py39haa95532_0
prompt-toolkit            3.0.36           py39haa95532_0
prompt_toolkit            3.0.36               hd3eb1b0_0
protego                   0.1.16                     py_0
psutil                    5.9.0            py39h2bbff1b_0
ptyprocess                0.7.0              pyhd3eb1b0_2
py                        1.11.0             pyhd3eb1b0_0
py-lief                   0.12.3           py39hd77b12b_0
pyasn1                    0.4.8              pyhd3eb1b0_0
pyasn1-modules            0.2.8                      py_0
pybind11-abi              4                    hd3eb1b0_0
pycodestyle               2.8.0              pyhd3eb1b0_0
pycosat                   0.6.4            py39h2bbff1b_0
pycparser                 2.21               pyhd3eb1b0_0
pyct                      0.5.0            py39haa95532_0
pycurl                    7.45.1           py39hcd4344a_0
pydispatcher              2.0.5            py39haa95532_2
pydocstyle                6.3.0            py39haa95532_0
pyerfa                    2.0.0            py39h2bbff1b_0
pyflakes                  2.4.0              pyhd3eb1b0_0
pygments                  2.11.2             pyhd3eb1b0_0
pyhamcrest                2.0.2              pyhd3eb1b0_2
pyjwt                     2.4.0            py39haa95532_0
pylint                    2.14.5           py39haa95532_0
pyls-spyder               0.4.0              pyhd3eb1b0_0
pynacl                    1.5.0            py39h8cc25b3_0
pyodbc                    4.0.34           py39hd77b12b_0
pyopenssl                 22.0.0             pyhd3eb1b0_0
pyparsing                 3.0.9            py39haa95532_0
pyqt                      5.15.7           py39hb08f45d_0    conda-forge
pyqt5-sip                 12.11.0          py39h415ef7b_0    conda-forge
pyqtwebengine             5.15.7           py39hb08f45d_0    conda-forge
pyrsistent                0.18.0           py39h196d8e1_0
pysocks                   1.7.1            py39haa95532_0
pytables                  3.7.0            py39h388bc9b_1
pytest                    7.1.2            py39haa95532_0
python                    3.9.16          h4de0772_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd3eb1b0_0
python-fastjsonschema     2.16.2           py39haa95532_0
python-libarchive-c       2.9                pyhd3eb1b0_1
python-lsp-black          1.2.1            py39haa95532_0
python-lsp-jsonrpc        1.0.0              pyhd3eb1b0_0
python-lsp-server         1.5.0            py39haa95532_0
python-slugify            5.0.2              pyhd3eb1b0_0
python-snappy             0.6.1            py39hd77b12b_0
python_abi                3.9                      2_cp39    conda-forge
pytoolconfig              1.2.5            py39haa95532_1
pytz                      2022.7           py39haa95532_0
pyviz_comms               2.0.2              pyhd3eb1b0_0
pywavelets                1.4.1            py39h2bbff1b_0
pywin32                   305              py39h2bbff1b_0
pywin32-ctypes            0.2.0           py39haa95532_1000
pywinpty                  2.0.2            py39h5da7b33_0
pyyaml                    6.0              py39h2bbff1b_1
pyzmq                     23.2.0           py39hd77b12b_0
qdarkstyle                3.0.2              pyhd3eb1b0_0
qstylizer                 0.2.2            py39haa95532_0
qt                        5.15.6               h91493d7_0    conda-forge
qt-main                   5.15.6               h9580fe5_6    conda-forge
qt-webengine              5.15.4               h0bdee70_3    conda-forge
qtawesome                 1.2.2            py39haa95532_0
qtconsole                 5.3.2            py39haa95532_0
qtpy                      2.2.0            py39haa95532_0
queuelib                  1.5.0            py39haa95532_0
regex                     2022.7.9         py39h2bbff1b_0
reproc                    14.2.4               hd77b12b_1
reproc-cpp                14.2.4               hd77b12b_1
requests                  2.28.1           py39haa95532_0
requests-file             1.5.1              pyhd3eb1b0_0
ripgrep                   13.0.0               haa95532_0
rope                      1.7.0            py39haa95532_0
rtree                     1.0.1            py39h2eaa2aa_0
ruamel.yaml               0.17.21          py39h2bbff1b_0
ruamel.yaml.clib          0.2.6            py39h2bbff1b_1
ruamel_yaml               0.17.21          py39h2bbff1b_0
s3transfer                0.6.0            py39haa95532_0
scikit-image              0.19.3           py39hd77b12b_1
scikit-learn              1.0.2            py39hf11a4ad_1
scikit-learn-intelex      2021.6.0         py39haa95532_0
scipy                     1.10.0           py39h321e85e_0
scrapy                    2.6.2            py39haa95532_0
seaborn                   0.12.2           py39haa95532_0
send2trash                1.8.0              pyhd3eb1b0_1
service_identity          18.1.0             pyhd3eb1b0_1
setuptools                65.6.3           py39haa95532_0
sip                       6.6.2            py39hd77b12b_0
six                       1.16.0             pyhd3eb1b0_1
smart_open                5.2.1            py39haa95532_0
snappy                    1.1.9                h6c2663c_0
sniffio                   1.2.0            py39haa95532_1
snowballstemmer           2.2.0              pyhd3eb1b0_0
sortedcollections         2.1.0              pyhd3eb1b0_0
sortedcontainers          2.4.0              pyhd3eb1b0_0
soupsieve                 2.3.2.post1      py39haa95532_0
sphinx                    5.0.2            py39haa95532_0
sphinxcontrib-applehelp   1.0.2              pyhd3eb1b0_0
sphinxcontrib-devhelp     1.0.2              pyhd3eb1b0_0
sphinxcontrib-htmlhelp    2.0.0              pyhd3eb1b0_0
sphinxcontrib-jsmath      1.0.1              pyhd3eb1b0_0
sphinxcontrib-qthelp      1.0.3              pyhd3eb1b0_0
sphinxcontrib-serializinghtml 1.1.5              pyhd3eb1b0_0
spyder                    5.3.3            py39hcbf5309_0    conda-forge
spyder-kernels            2.3.3            py39haa95532_0
sqlalchemy                1.4.39           py39h2bbff1b_0
statsmodels               0.13.5           py39h080aedc_1
sympy                     1.11.1           py39haa95532_0
tabulate                  0.8.10           py39haa95532_0
tbb                       2021.6.0             h59b6b97_1
tbb4py                    2021.6.0         py39h59b6b97_1
tblib                     1.7.0              pyhd3eb1b0_0
tenacity                  8.0.1            py39haa95532_1
terminado                 0.17.1           py39haa95532_0
testpath                  0.6.0            py39haa95532_0
text-unidecode            1.3                pyhd3eb1b0_0
textdistance              4.2.1              pyhd3eb1b0_0
threadpoolctl             2.2.0              pyh0d69192_0
three-merge               0.1.1              pyhd3eb1b0_0
tifffile                  2021.7.2           pyhd3eb1b0_2
tinycss2                  1.2.1            py39haa95532_0
tk                        8.6.12               h2bbff1b_0
tldextract                3.2.0              pyhd3eb1b0_0
toml                      0.10.2             pyhd3eb1b0_0
tomli                     2.0.1            py39haa95532_0
tomlkit                   0.11.1           py39haa95532_0
toolz                     0.12.0           py39haa95532_0
tornado                   6.2              py39h2bbff1b_0
tqdm                      4.64.1           py39haa95532_0
traitlets                 5.7.1            py39haa95532_0
twisted                   22.2.0           py39h2bbff1b_1
twisted-iocpsupport       1.0.2            py39h2bbff1b_0
typing-extensions         4.4.0            py39haa95532_0
typing_extensions         4.4.0            py39haa95532_0
tzdata                    2022g                h04d1e81_0
ucrt                      10.0.20348.0         haa95532_0
ujson                     5.4.0            py39hd77b12b_0
unidecode                 1.2.0              pyhd3eb1b0_0
urllib3                   1.26.14          py39haa95532_0
vc                        14.2                 h21ff451_1
vs2015_runtime            14.34.31931         h4c5c07a_10    conda-forge
w3lib                     1.21.0             pyhd3eb1b0_0
watchdog                  2.1.6            py39haa95532_0
wcwidth                   0.2.5              pyhd3eb1b0_0
webencodings              0.5.1            py39haa95532_1
websocket-client          0.58.0           py39haa95532_4
werkzeug                  2.2.2            py39haa95532_0
whatthepatch              1.0.2            py39haa95532_0
wheel                     0.38.4           py39haa95532_0
widgetsnbextension        3.5.2            py39haa95532_0
win_inet_pton             1.1.0            py39haa95532_0
win_unicode_console       0.5              py39haa95532_0
wincertstore              0.2              py39haa95532_2
winpty                    0.4.3                         4
wrapt                     1.14.1           py39h2bbff1b_0
xarray                    2022.11.0        py39haa95532_0
xlrd                      2.0.1              pyhd3eb1b0_0
xlsxwriter                3.0.3              pyhd3eb1b0_0
xlwings                   0.27.15          py39haa95532_0
xz                        5.2.10               h8cc25b3_1
yaml                      0.2.5                he774522_0
yaml-cpp                  0.7.0                hd77b12b_1
yapf                      0.31.0             pyhd3eb1b0_0
zeromq                    4.3.4                hd77b12b_0
zfp                       0.5.5                hd77b12b_6
zict                      2.1.0            py39haa95532_0
zipp                      3.11.0           py39haa95532_0
zlib                      1.2.13               hcfcfb64_4    conda-forge
zope                      1.0              py39haa95532_1
zope.interface            5.4.0            py39h2bbff1b_0
zstandard                 0.18.0           py39h2bbff1b_0
zstd                      1.5.2                h19a0ad4_0

Now that you know a little about me, let me tell you about the issue I am
having:

# Description of Issue
i ran this:
import os
import zipline
from zipline.utils.run_algo import load_extensions
from zipline.data import bundles
from norgatedata import StockPriceAdjustmentType
from zipline_norgatedata import (
    register_norgatedata_equities_bundle,
    register_norgatedata_futures_bundle )
load_extensions(
    default=True,
    extensions=[],
    strict=True,
    environ=os.environ,
)

bundles.ingest('norgatedata-fang', show_progress = False)

* What did you expect to happen? 
    Correct Ingestion of registered bundle
* What happened instead?
    [2023-02-19 11:02:13.563293] INFO: Norgate Data: NorgateData package v1.0.65: Init complete
[2023-02-19 11:02:13.568369] INFO: Norgate Data: Zipline v2.2.0 detected
[2023-02-19 11:02:13.568369] INFO: Norgate Data: Zipline_norgatedata package v2.3.3: Init complete
[2023-02-19 11:02:14.985334] INFO: zipline.data.bundles.core: Ingesting norgatedata-fang.
[2023-02-19 11:02:14.985334] INFO: Norgate Data: Ingesting equities bundle norgatedata-fang with start date 2012-05-18
[2023-02-19 11:02:14.995877] INFO: Norgate Data: Obtaining metadata for 5 securities...
[2023-02-19 11:02:15.037736] INFO: Norgate Data: Metadata process complete.  Revised security count: 5
Traceback (most recent call last):
  File "C:\Users\markg\anaconda3\envs\zip38\lib\site-packages\zipline\data\bundles\core.py", line 445, in ingest
    bundle.ingest(
  File "C:\Users\markg\anaconda3\envs\zip38\lib\site-packages\zipline_norgatedata\zipline_norgatedata.py", line 206, in ingest
    asset_db_writer.write(equities=metadata, exchanges=exchanges)
  File "C:\Users\markg\anaconda3\envs\zip38\lib\site-packages\zipline\assets\asset_writer.py", line 811, in write
    self._real_write(
  File "C:\Users\markg\anaconda3\envs\zip38\lib\site-packages\zipline\assets\asset_writer.py", line 487, in _real_write
    self.init_db(conn)
  File "C:\Users\markg\anaconda3\envs\zip38\lib\site-packages\zipline\assets\asset_writer.py", line 912, in init_db
    tables_already_exist = self._all_tables_present(txn)
  File "C:\Users\markg\anaconda3\envs\zip38\lib\site-packages\zipline\assets\asset_writer.py", line 888, in _all_tables_present
    conn = txn.connect()
AttributeError: 'Connection' object has no attribute 'connect'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "g:/My Drive/Projects/Notebooks/zipline/zipline_ingest.py", line 18, in <module>
    bundles.ingest('norgatedata-fang', show_progress = False)
  File "C:\Users\markg\anaconda3\envs\zip38\lib\site-packages\zipline\data\bundles\core.py", line 469, in ingest
    downgrade(wf.path, version)
  File "C:\Users\markg\anaconda3\envs\zip38\lib\contextlib.py", line 525, in __exit__
    raise exc_details[1]
  File "C:\Users\markg\anaconda3\envs\zip38\lib\contextlib.py", line 510, in __exit__
    if cb(*exc_details):
  File "C:\Users\markg\anaconda3\envs\zip38\lib\site-packages\zipline\utils\cache.py", line 385, in __exit__
    rmtree(self.path)
  File "C:\Users\markg\anaconda3\envs\zip38\lib\shutil.py", line 740, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "C:\Users\markg\anaconda3\envs\zip38\lib\shutil.py", line 613, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "C:\Users\markg\anaconda3\envs\zip38\lib\shutil.py", line 613, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "C:\Users\markg\anaconda3\envs\zip38\lib\shutil.py", line 618, in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
  File "C:\Users\markg\anaconda3\envs\zip38\lib\shutil.py", line 616, in _rmtree_unsafe
    os.unlink(fullname)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\markg\\AppData\\Local\\Temp\\tmpbzksn37a\\tmp029o9sqq\\norgatedata-fang\\2023-02-19T11;02;14.903658\\assets-7.sqlite'
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "C:\Users\markg\anaconda3\envs\zip38\lib\shutil.py", line 616, in _rmtree_unsafe
    os.unlink(fullname)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\markg\\AppData\\Local\\Temp\\tmpbzksn37a\\tmp029o9sqq\\norgatedata-fang\\2023-02-19T11;02;14.903658\\assets-7.sqlite'

Here is how you can reproduce this issue on your machine:

## Reproduction Steps

1.Load code as above
2.Run code
3.
...

## What steps have you taken to resolve this already?
Searched all github issues and google but not found similar issue

...

# Anything else?

I have tried closing every program that has anything to do with zipline being open,
including explorer, jupyter etc

I have also tried saving the py file locally and running from there (not google drive) but same error occurs

...

Sincerely,
windowspc\markg

**Comment by MBounouar** [comment]
Hi,
I have no idea what's the real issue here (maybe Windows or the python context manager doesn't work properly) .
For some reason windows doesn't allow the folder to be moved from the temporary directory to the final location.

I have seen it happen once on a windows machine installation, but unfortunately can't replicate it, as I don't have a windows machine at hand.

I can only suggest to maybe try to use a python 3.9 environment to see if the error persists.
Or, if there is data under the temporary norgatedata-fang\2023-02-19T11;02;14.903658
to simply move it you the zipline/data folder

Cheers,
MBounouar

**Comment by markgahagan1** [comment]
Thanks for coming back to me.
I tried completely uninstalling anaconda, installing miniconda and creating a python 3.9 environment. I then refollowed the instructions to install zipline reloaded and zipline norgatedata. Unfortunately exactly the same error occurs under 3.9. As you suggested there is data in the Temp folder which is not being moved to the correct folder. So the issue is somewhere in between the two operations.
